modules/300-prometheus/images/alerts-receiver/src/server.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2023 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"net"
+	"net/http"
+	"sync/atomic"
+
+	"github.com/prometheus/common/model"
+	log "github.com/sirupsen/logrus"
+)
+
+type serverStruct struct {
+	*http.Server
+	isReady atomic.Bool
+}
+
+func (s *serverStruct) readinessHandler(w http.ResponseWriter, _ *http.Request) {
+	if !s.isReady.Load() {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (s *serverStruct) livenessHandler(w http.ResponseWriter, _ *http.Request) {
+	w.WriteHeader(http.StatusOK)
+}
+
+func newServer(host, port string) *serverStruct {
+	return &serverStruct{
+		Server: &http.Server{
+			Addr: net.JoinHostPort(host, port),
+		},
+	}
+}
+
+func (s *serverStruct) setHandlers(config *configStruct, store *storeStruct) {
+	http.HandleFunc("/healthz", s.livenessHandler)
+	http.HandleFunc("/readyz", s.readinessHandler)
+	http.HandleFunc("/api/v1/alerts", s.alertsHandler(config, store))
+	http.HandleFunc("/api/v2/alerts", s.alertsHandler(config, store))
+}
+
+func (s *serverStruct) ready() {
+	s.isReady.Store(true)
+}
+
+func (s *serverStruct) notReady() {
+	s.isReady.Store(false)
+}
```

Comment:
```suggestion
func (s *serverStruct) ready() {
	s.isReady.Store(true)
}

func (s *serverStruct) notReady() {
	s.isReady.Store(false)
}
```

I guess you can handle this with a single function like:
```go
func (s *serverStruct) setReadiness(ready bool) {
	s.isReady.Store(ready)
}
```
instead of 2 functions